src/Connector/BaseApiConnector.php

```diff
@@ -36,8 +36,7 @@ public function resolveResponse(Response $response)
         $objectName = $this->getObjectName();
 
         /** @var BaseApiObject $object */
-        $object = new $objectName();
-        $object->setData($content);
+        $object = new $objectName($content);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$object". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=php%3AS1488)